### PR TITLE
added to binary search to reduce computation from O(N) to O(logN)

### PR DESCRIPTION
**Pull request by rahul1809**
Modified the program logic to reduce complexity from linear to logarithmic in time.

### CODE REVIEW

File: code/mathematical-algorithms/Check_is_square/FindingISquare.cpp [hunk 1]
```diff
@@ -1,25 +1,51 @@
-#include <iostream>
+#include <bits/stdc++.h>
 using namespace std;
-int main()
+
+typedef long long int lld;
```
Comment (arnavb) [inline code/mathematical-algorithms/Check_is_square/FindingISquare.cpp]: Why the _d_ at the end. Why not _i_?

File: code/mathematical-algorithms/Check_is_square/FindingISquare.cpp [hunk 2]
```diff
@@ -1,25 +1,51 @@
...
+      mid = min + (max - min)/2;
+    }
+    else if (mid*mid < number)
+    {
+      min = mid + 1;
+      mid = min + (max - min)/2;
+
+    }
+    if (mid*mid == number)
+    return true;
```
Comment (arnavb) [inline code/mathematical-algorithms/Check_is_square/FindingISquare.cpp]: Add a tab for clarity

File: code/mathematical-algorithms/Check_is_square/FindingISquare.cpp (hunk 1 above)
Comment (rahul1809) [reply code/mathematical-algorithms/Check_is_square/FindingISquare.cpp]: In C, we have 'lld' as format specifier for long long int . I'm just sticking to that convention of naming.

File: code/mathematical-algorithms/Check_is_square/FindingISquare.cpp (hunk 2 above)
Comment (rahul1809) [reply code/mathematical-algorithms/Check_is_square/FindingISquare.cpp]: added tabs for clarity

**Comment by AdiChat** [comment]
Kindly do not delete the existing code `FindingISquare.cpp` as it is using a different approach 👍

**Comment by rahul1809** [comment]
Ok will do that in future. Add new and better approaches in seperate files.

**Pull request closed without merging**